components/brave_federated/task/typing.h

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_FEDERATED_TASK_TYPING_H_
+#define BRAVE_COMPONENTS_BRAVE_FEDERATED_TASK_TYPING_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "brave/components/brave_federated/task/model.h"
+#include "brave/components/brave_federated/util/linear_algebra_util.h"
+
+namespace brave_federated {
+
+enum class TaskType { kEvaluation, kTraining, kUndefined };
+
+struct TaskId {
+  std::string id;
+  std::string group_id;
+  std::string family_id;
+
+  bool IsValid() const {
```

Comment:
nit: `[[nodiscard]]` for IsValid calls, and any other calls where the return value if ignored would likely cause bugs.

Reply:
I don't see it used anywhere in the brave codebase, except for a test. Do I just prepend it to the definition/implementation?

Reply:
definition. up to you. but it just means nobody can discard the value and as it is check validity this is a good place to use